openmc/geometry.py

```diff
@@ -409,14 +446,21 @@ def get_redundant_surfaces(self):
             that should replace it.
 
         """
-        tally = defaultdict(list)
-        for surf in self.get_all_surfaces().values():
-            coeffs = tuple(surf._coefficients[k] for k in surf._coeff_keys)
-            key = (surf._type,) + coeffs
-            tally[key].append(surf)
-        return {replace.id: keep
-                for keep, *redundant in tally.values()
-                for replace in redundant}
+        # check if redundant surfaces have not been calculated yet
+        if self._redundant_surface_map is None:
+            tally = defaultdict(list)
+            for surf in self.get_all_surfaces().values():
+                coeffs = tuple(round(surf._coefficients[k],
+                                     self.surface_precision)
+                               for k in surf._coeff_keys)
+                key = (surf._type,) + coeffs
+                tally[key].append(surf)
+
+            self._redundant_surface_map = {replace.id: keep
+                                           for keep, *redundant in tally.values()
+                                           for replace in redundant}
```

Comment:
```suggestion
            redundancies = defaultdict(list)
            for surf in self.get_all_surfaces().values():
                coeffs = tuple(round(surf._coefficients[k],
                                     self.surface_precision)
                               for k in surf._coeff_keys)
                key = (surf._type,) + coeffs
                redundancies[key].append(surf)

            self._redundant_surface_map = {replace.id: keep
                                           for keep, *redundant in redundancies.values()
                                           for replace in redundant}
```

Just avoiding the tally name here so the meaning of the name isn't confused with the `openmc.Tally` object.